daemon/src/main/java/eu/maveniverse/maven/mimir/daemon/Daemon.java

```diff
@@ -298,6 +299,10 @@ public Daemon(
             // we are done
         } catch (Exception e) {
             logger.error("Error while accepting client connection", e);
+        } finally {
+            Files.deleteIfExists(daemonConfig.socketPath());
```

Comment:
`deleteIfExists` may throw an exception, which not cause the `daemonLockDir` to not be unlocked

Reply:
```
try {
    Files.deleteIfExists(daemonConfig.socketPath());
} finally {
    Files.deleteIfExists(daemonConfig.socketPath());
}
```